File: Day-06/day_06.py

def sort(testdata):

    red=0
    white=0
    blue=0
    
    for i in range(len(testdata)):
        if(testdata[i]==0):
            red+=1
        elif(testdata[i]==1):
            white+=1
        else:
            blue+=1
    for i in range(len(testdata)):
        if(i<red):
            testdata[i]= 0
        elif(i<red+white):
            testdata[i]=1
        else:
            testdata[i]=2   
    return testdata     
        
test_case_1 = [2,0,2,1,1,0]
test_case_2 = [2,0,1]


print("--- test case 1 ---");
print(f"Input: nums = {test_case_1}\nOutput: {sort(test_case_1)}")
print("--- test case 2 ---")
print(f"Input: nums = {test_case_2}\nOutput: {sort(test_case_2)}")

File: Day-06/day_06.java
import java.util.*;

class Future{

    static int[] sort(int arr[]){

        int red=0,white=0,blue=0;

        for(int i=0 ;i <arr.length ;i++){

                if(arr[i]==0){
                    red++;
                }else if(arr[i]==1){

                    white++;
                }else{
                    blue++;
                }
        }

        for(int i=0 ;i <arr.length ;i++){

            if(i<red){

                arr[i]= 0;

            }else if(i<red+white){
                arr[i]=1;

            }else{
                arr[i]=2;
            }
        }
        return arr;
    }
    public static void main(String[] args) {

        int test_case1[] = new int[] {2,0,2,1,0,1};
        int test_case2[] = new int[] {2,0,1};

        int output1[] = sort(test_case1);
        int output2[] = sort(test_case2);

        System.out.println(Arrays.toString(test_case1) + " Output is "+Arrays.toString(output1));
        System.out.println(Arrays.toString(test_case2) + " Output is "+Arrays.toString(output2));

    }
}